server/events/git_cred_writer.go

```diff
@@ -42,10 +43,16 @@ func WriteGitCreds(gitUser string, gitToken string, gitHostname string, home str
 
 	logger.Info("wrote git credentials to %s", credsFile)
 
-	cmd := exec.Command("git", "config", "--global", "credential.helper", "store")
-	if out, err := cmd.CombinedOutput(); err != nil {
-		return errors.Wrapf(err, "There was an error running %s: %s", strings.Join(cmd.Args, " "), string(out))
+	credentialCmd := exec.Command("git", "config", "--global", "credential.helper", "store")
+	if out, err := credentialCmd.CombinedOutput(); err != nil {
+		return errors.Wrapf(err, "There was an error running %s: %s", strings.Join(credentialCmd.Args, " "), string(out))
+	}
+	logger.Info("successfully ran %s", strings.Join(credentialCmd.Args, " "))
+
+	urlCmd := exec.Command("git", "config", "--global", fmt.Sprintf("url.https://%s@%s.insteadOf", gitUser, gitHostname), fmt.Sprintf("ssh://git@%s", gitHostname))
```

Comment:
This didn't work for me. I needed to add the token. Did it work for you?
```
https://user:token@hostname
```

Reply:
Yeah, I have it working with just `https://user@hostname`

Reply:
@lkysow Sorry about the radio silence I was on holiday. Did you want me to update it to be `https://user:token@hostname`? I have it working currently with just `https://user@hostname` but that's GitHub perhaps it needs to be more specific for some of the other providers.

Reply:
No worries, I don't know if it needs to be updated. I just tried it out and it wasn't working without that. I used GitHub too. My next step was to try it out again and confirm whether if was/wasn't working.

Reply:
I'm not sure how it would work without that though. Why would github let you clone a private repository with just a username?

Reply:
Excuse my eavesdropping :) but thought I could chime in here and shed some light.

Github allows using a token and empty string as a valid username/password pair. For example:

```bash
$ git clone "https://${GITHUB_TOKEN}:@github.com/mikedougherty/dotfiles.git"
Cloning into 'dotfiles'...
remote: Enumerating objects: 721, done.
remote: Total 721 (delta 0), reused 0 (delta 0), pack-reused 721
Receiving objects: 100% (721/721), 2.01 MiB | 6.50 MiB/s, done.
Resolving deltas: 100% (161/161), done.
```
This also works with their API:

```bash
$ curl -su "${GITHUB_TOKEN}:" https://api.github.com/user | jq -r .login
mikedougherty
```

I've always thought this was a strange design choice on their part but it does work without issue.

Reply:
> I'm not sure how it would work without that though. Why would github let you clone a private repository with just a username?

I don't believe it is. Keep in mind that the `.git-credentials` file also exists which has the credentials git needs to authenticate as the relevant user. I'm guessing that's how it's working...

Reply:
Any progress on this ? I've tested this via a custom docker container that generated the file like this: 
```
echo -e $'[url "https://${ATLANTIS_GH_USER}@github.com"] \n\
   insteadOf = ssh://git@github.com'\
 >> "${HOME}/.gitconfig" \
```

and seems to be working. So I guess @ImperialXT is right about this